cmd/zar/README.md

```diff
@@ -120,71 +155,82 @@ If you want to see one, just look at it with zq, e.g.
 instead of the logs and run much faster...
 ```
 zar find :ip=10.10.23.2
 ```
+In the output here, you'll see this IP exists in exactly one log file:
+```
+/path/to/ZAR_ROOT/20180324/1521912868.861247.zng
+```
 
-### micro-indexes
+## micro-indexes
 
-We call these files "micro indexes" because each index pertains to just one
-chunk of log file and represents just one indexing rule.
+We call these zng files "micro indexes" because each index pertains to just one
+chunk of log file and represents just one indexing rule.  If you're curious about
+what's in the index, it's just a sorted list of keyed records along with some
+additional zng streams that comprise a constant b-tree index into the sorted list.
+But the cool thing here is that everything is just a zng stream.
 
-We're not building a massive, integrated inverted index that
-tells you exactly where each event is in the event store.  Our model is to
-instead build lots of small indexes for each log chunk and index different things
-in the different indexes.
+Instead of building a massive, inverted index with glorious roaring
+bitmaps that tell you exactly where each event is in the event store, our model
+is to instead build lots of small indexes for each log chunk and index different
+things in the different indexes.
 
-### creating more micro-indexes
+## creating more micro-indexes
 
-You can add and delete micro-indexes whenever you want.  Let's say you later
-decide you want searches over the uri field to run fast.  You just run
-"zar index" again:
+The beauty of this approach is that you can add and delete micro-indexes
+whenever you want.  No need to suffer the fate of a massive reindexing
+job when you have a new idea about what to index.
+
+So, let's say you later decide you want searches over the "uri" field to run fast.
+You just run "zar index" again but with different parameters:
 ```
 zar index uri
 ```
-No need to run a massive re-indexing job if you change the indexing rules
-since each micro-index is independent of the other.
-
 And now you can run field matches on `uri`:
 ```
 zar find uri=/file
 ```
-
+and you'll get
+```
+/path/to/ZAR_ROOT/20180324/1521911720.600725.zng
+/path/to/ZAR_ROOT/20180324/1521912191.526264.zng
+```
 If you have a look, you'll see there are index files now for both type ip
 and field uri:
 ```
 zar ls -l
 ```
 
-### operating directly on micro-indexes
+## operating directly on micro-indexes
 
 Let's say instead of searching for what log chunk a value is in, we want to
-actually pull out the zng records that comprise the index.  The syntax
-is kind of clunky and we're working to clean it up but you can say...
+actually pull out the zng records that comprise the index.  This turns out
+to be really powerful in general, but to give you a taste here, you can say...
 ```
-zar find -z -o - -x zdx:type:ip 10.47.21.138 | zq -t -
+zar find -z -x zdx:type:ip 10.47.21.138 | zq -t -
```

Comment:
@mccanne this example introduces both the `-z` and the `-x` flags, but only `-z` is described in the paragraph below below. Can the example just be `zar find -z :ip=10.47.21.138 | zq -t -` ?